### PR TITLE
Autonomous Vehicle Platforms Literature Review and Caffe Setup

### PR DESCRIPTION
**Pull request by KushRJ**

### CODE REVIEW

**Comment by srija-ieee** [comment]
Thank you for the commits Kush. Apparently 1 active pull request per branch is sufficient. In this case we are working off the master branch so thats already open.

File: Kush/caffe_installation_updated [hunk 1]
```diff
@@ -0,0 +1,35 @@
+sudo apt-get install -y --no-install-recommends libboost-all-dev
```
Comment (srija-ieee) [inline Kush/caffe_installation_updated:1]: Avoid no install recommends unless specific reason to do so

File: Kush/caffe_installation_updated [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+sudo apt install python-pip
+sudo apt install git
+git clone https://github.com/BVLC/caffe
+cd caffe
+cp Makefile.config.example Makefile.config
+
+sudo pip install scikit-image protobuf
+cd python
+for req in $(cat requirements.txt); do sudo pip install $req; done
```
Comment (srija-ieee) [inline Kush/caffe_installation_updated:13]: Where is requirements.txt?

File: Kush/caffe_installation_updated [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+cd ..
+sudo apt install vim
+sudo vim Makefile.config
+#TODO: Press 'i' on keyboard to insert in the Makefile
```
Comment (srija-ieee) [inline Kush/caffe_installation_updated:18]: Pls use sed to automate text replacement

File: Kush/caffe_installation_updated [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+#      CPU-only switch (uncomment to build without GPU support).
+#      Uncomment if you're using OpenCV 3
+#      Also add the following path in INCLUDE_DIRS := /usr/include/hdf5/serial
+#      Press 'ESC' to exit, ':w' and ':x' to save and exit Makefile
+
+find /usr -iname "*hdf5.h*"
```
Comment (srija-ieee) [inline Kush/caffe_installation_updated:24]: What is the use of this in a script?

File: Kush/caffe_installation_updated [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+export CPATH="/usr/include/hdf5/serial"
```
Comment (srija-ieee) [inline Kush/caffe_installation_updated:25]: Setting CPATH to specific driver folder during compilation can lead to other failures.

To provide a specific path to the gcc build step please use the gcc -I command line option

File: Kush/caffe_installation_updated [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+#      Uncomment if you're using OpenCV 3
+#      Also add the following path in INCLUDE_DIRS := /usr/include/hdf5/serial
+#      Press 'ESC' to exit, ':w' and ':x' to save and exit Makefile
+
+find /usr -iname "*hdf5.h*"
+export CPATH="/usr/include/hdf5/serial" 
+make all
+
+cd /usr/lib/x86_64-linux-gnu/
+sudo ln -s /usr/lib/x86_64-linux-gnu/libhdf5_serial.so.10.1.0 /usr/lib/x86_64-linux-gnu/libhdf5.so
```
Comment (srija-ieee) [inline Kush/caffe_installation_updated:29]: Avoid creating links and try with the LD_LIBRARY_PATH variable

**Review by srija-ieee — CHANGES_REQUESTED** [review]

**Comment by KushRJ** [comment]
Ma'am I didn't use requirements.txt, so I didn't check for it.
I didn't know about the command sed, so I used vim and gave instructions on
how to modify makefile accordingly.

> +
+sudo pip install scikit-image protobuf
+cd python
+for req in $(cat requirements.txt); do sudo pip install $req; done
+
+cd ..
+sudo apt install vim
+sudo vim Makefile.config
+#TODO: Press 'i' on keyboard to insert in the Makefile
+#      CPU-only switch (uncomment to build without GPU support).
+#      Uncomment if you're using OpenCV 3
+#      Also add the following path in INCLUDE_DIRS := /usr/include/hdf5/serial
+#      Press 'ESC' to exit, ':w' and ':x' to save and exit Makefile
+
+find /usr -iname "*hdf5.h*"
+export CPATH="/usr/include

I put this because I was getting errors of hdf5 library.
I don't know how to use the gcc command so I didn't use it.

**Comment by srija-ieee** [comment]
Please see the following script
```
#!/bin/bash
#Todo: Add license info (GPL v2)
#Todo: Add copyright notice – dual copyright AyeAI and authorization
#Todo: Add change info with author name

#Install default version on caffe distributed on Ubuntu
sudo apt install caffe-cpu

#Build caffe from Ubuntu sources
sudo apt build-dep caffe-cpu
#Caffe CPUONLY uses ATLAS for BLAS
sudo apt install libatlas-base-dev

apt-get source  caffe-cpu
cd caffe-1.0.0/

#Use sed or awk to modify CmakeLists.txt to
# - turn on CPUONLY, turn off rest
cp CMakeLists.txt CMakeLists.txt.orig
cat CMakeLists.txt.orig | awk '/CPU_ONLY/ {if ($7 == "OFF)") $7 = "ON)"};/./ {print $0}' > CMakeLists.txt

mkdir build
cd build
cmake ..
make all
#The following is a brute force for now and may 
sudo make install
make runtest
```

**Comment by obonac** [comment]
Dear Kush,

Congratulations on completing a month long summer internship with AyeAI Consulting, on our Autonomous Ambulance project during July 2019! We have thoroughly enjoyed working with you and are proud to have you among our alumni.

We thank you for the effort and appreciate your perseverance and dedication to the tasks assigned. You have demonstrated a very good ability to grasp new concepts and have been good at working with remote working methods, maintaining perfect availability and responsiveness.

We believe the exposure to "image recognition models and traffic light detection system", virtualization, automating installation of frameworks / models / datasets, running inference through shell scripts, running AI models without GUI, managing library dependencies, and doing root cause analysis to get a system to work will give you a solid foundation for your future Machine Learning / Deep Learning projects on IoT. We expect you will continue to build further depth in the topics learnt in correlation with your studies and coursework.

We look forward to your continued interest in the domain and participation in various open source projects including those from other verticals of AyeAI. We also invite you to participate in our Campus Ambassadors Program (AyeCAP) wherein you can help spread awareness of AI, IoT and industry methodologies among your friends and colleagues.

We would like to gently remind you of standing obligations with regards to intellectual property as accepted during initial documentation; Whereas, all due diligence have been exercised in engaging you only in IP neutral open source projects, Notwithstanding, given the complex nature of technical environments, there may have been inadvertent exposure to some proprietary material or official secrets such as, and Not limited to, product or service roadmaps, undisclosed inventions, unpublished material or other intellectual property and, Therefore, we would advise you to keep us informed and consult with our legal advisors in case you are involved in projects similar to the AyeAI offerings in the near future till a period of two years from the date of completion of your latest internship, employment, or other engagement with us. We would also like to mention that any usage of the AyeAI trademark, or affiliated organizations' logos, trademarks, copyrighted material, or any other intellectual property requires explicit permission from the respective company leadership. We will be happy to provide you appropriate certificates and references related to the work done. For all official communications please use the contact links available from our website with your AyeAI Unique ID which shall be communicated to you shortly. Kindly do not expose any Identifiable or Sensitive Personal Information (ISPI) on public platforms and share those only over secure and private channels. You are encouraged to actively bring to our notice any issues with legal and compliance that you encounter in the future related to your engagement with us.

This letter marks the full and final closure of the current engagement with AyeAI Consulting, with no further dues or actionable items remain pending on either party. There is no objection to your participation in any other internship or employment hereafter.

Wishing you all the very best in your academics and looking forward to having you in the AyeCAP program. 

Regards,
Abhishek Choudhary
Founder - AyeAI

PS: The pull request is approved pertaining to review by legal team prior to inclusion in any AyeAI deliverables.

Note: Valid only when committed under the official account of the author. No warranties or fitness is guaranteed on the deliverables. The content has not been endorsed and the original commiting authors are solely responsible for ensuring no intellectual property have been infringed. Accepted in good faith only.

**Pull request merged**